sdk/keyvault/keyvault-certificates/samples/javascript/mergeCertificate.js

```diff
@@ -27,7 +27,7 @@ async function main() {
 
   // Retrieving the certificate's signing request
   const operationPoller = await client.getCertificateOperation("MyCertificate");
-  const { csr } = operationPoller.getResult();
+  const { csr } = operationPoller.getPublicState().operation;
```

Comment:
so the public state contains an operation which is exactly the same shape as the result type?

Reply:
not at all, the result type is KeyVaultCertificateWithPolicy and the operation is CertificateOperation.

Reply:
I guess I was confused by the previous signature: 

```
getCertificateOperation(certificateName: string, options?: GetCertificateOperationOptions): Promise<PollerLike<PollOperationState<CertificateOperation>, CertificateOperation>>;
```

Since `PollerLike<TState, TResult>` means that `CertificateOperation` was `TResult`, no?

Reply:
yep! TResult was CertificateOperation. If that was the case, the previous code would be sufficient. The issue arises when, during the alignment across languages, it appears evident that there's the need to expose other information than the result type, since other languages had CertificateWithPolicy as the result of the poller, but they also allowed users to obtain the certificate operation in another way.